src/base/bittorrent/session.cpp

```diff
@@ -1954,6 +1978,17 @@ bool Session::addTorrent_impl(CreateTorrentParams params, const MagnetUri &magne
     p.flags &= ~lt::torrent_flags::duplicate_is_error; // Already checked
 #endif
 
+    // Limits
+    p.max_connections = maxConnectionsPerTorrent();
```

Comment:
You break it here.
Currently we do this unconditionally since libtorrent will override all these fields from fastresume data (if it's provided). In case of libtorrent-1.2 you should do it only for new torrents (there is nothing wrong if you will do it in both libtorrent version cases).

Reply:
>there is nothing wrong if you will do it in both libtorrent version cases

This is already true isn't? Or I should really move it to new torrents when using libtorrent 1.2?

Reply:
>This is already true isn't?

No, of course. Otherwise why I say that you break it?
You should assign all these default values only for new torrents (at least in case of libtorrent-1.2).

Reply:
>You break it here.

Can you explain it a bit? What did you observe?
Moving these code to new torrent only will break other things.

Reply:
Current (valid) behavior:
1. Fill add_torrent_params with default values.
2. If fastresume data is presented libtorrent will use it to override the values from add_torrent_params

Your (wrong in case of libtorrent-1.2) behavior:
1. Fill add_torrent_params from fastresume data
2. Override add_torrent_params fields with default values.

You should prefer fastresume data if it's available (i.e. in case of restored torrent) and it isn't corrupted.
>Moving these code to new torrent only will break other things.

What will be broken by this action?

Reply:
>You should prefer fastresume data if it's available (i.e. in case of restored torrent) and it isn't corrupted.

Thanks, I understand this part now.

>What will be broken by this action?

When there is a magent torrent and its metadata hasn't fetched yet, it still needs to setup the default parameters. In this case `!fromMagnetUri && params.restored` will be true and the fastresume file will only contain "qbt-*" entries.

I've pushed an updated version.